### ISSUE TITLE
a11y: Fix keyboard navigation, labels/descriptions, roles and focus display for workload health (scale up/down) control

### ISSUE THREAD

**Issue by nwmac** [rancher/dashboard#12782]
a11y Issue: 161, 159, 160

**Issue 161 - The workload health control is not accessible with the keyboard.**

![image](https://github.com/user-attachments/assets/7d904ab9-38fc-40f4-84c2-4f60720c24fc)

**Issue 159 - Role and state not defined** 

The role and state are not programmatically defined for the interactive image making it difficult for screen reader users to access the state information and functionality associated with interactive image. 

![image](https://github.com/user-attachments/assets/d15cd8a7-0005-49da-802e-e37858a59eca)

Suggested remediation:

Apply the following changes:
• Code the mentioned interactive images using <button> tag. Alternatively, use role="button" and tabindex="0" attributes to define the button role.
• Modify the scripts to ensure that the functionality is accessible for all users.
• Set "aria-expanded" attribute to "false" for the image buttons when the buttons are in the collapsed state. Ensure that the value of "aria-expanded" attribute changes on user interaction via scripting.

**Issue 160 - Textual description missing for interactive images** 

The mentioned interactive images lack textual description. Moreover, they are defined through CSS background property resulting in screen reader users will be unable to identify the purpose of these interactive images.

![image](https://github.com/user-attachments/assets/e72e7edf-9b91-4117-9090-3dd76471cc23)

Suggested remediation:

Implement the Health progress images using HTML <img> element and provide alternate text using "alt" attribute in its source code.

Alternatively, use CSS content property (with ::before and ::after pseudo-elements) to define the images. Specify textual description using visually hidden text. If hidden text is not possible, then use the "aria-label" attribute.

**Comment by IsaSih** [comment]
Keyboard navigation with enter/return works as expected in 
Rancher | v2.11-3d729db8a848fb2523bbc2f3f21aa5367dcdcc69-head
Dashboard | master df18acc
Rancher CLI | v2.11.0-rc.1

However, the space bar opens the health control bar, but also performs the undesired action to scroll down the page.

https://github.com/user-attachments/assets/43cc351e-4d81-435b-ae6a-a243f8b97003

**Comment by aalves08** [comment]
@IsaSih , I've created a PR https://github.com/rancher/dashboard/pull/13413 to fix the `space` scroll issue.

As for:

> From the issue scope
> 
> ```
> Suggested remediation:
> 
> Implement the Health progress images using HTML  element and provide alternate text using "alt" attribute in its source code.
> 
> Alternatively, use CSS content property (with ::before and ::after pseudo-elements) to define the images. Specify textual description using visually hidden text. If hidden text is not possible, then use the "aria-label" attribute.
> ```
> 
> Was this implemented? If that's the case how should this be tested?

I've added an `aria-label` with the description of the statuses of the workload health scale (progress bar). It's the solution that fits the bill with the least amount of changes to the component. We don't want to introduce more work with a lot still to do 🙏 

<img width="2323" alt="Image" src="https://github.com/user-attachments/assets/e5350f2e-8829-4dbd-ac92-12ac67042847" />
<img width="2323" alt="Image" src="https://github.com/user-attachments/assets/b175e5fa-5da2-441a-8af1-caec6bb5d2be" />

**Comment by IsaSih** [comment]
Space bar behavior works as expected. Tests pass in 
Rancher | v2.11-1ec5c626b1f615573b7f7269d4eb67ddb8c82749-head
Dashboard | master c15d79e
Rancher CLI | v2.11.0-rc.1
